pkg/kmodule/kmodule_linux.go

```diff
@@ -103,6 +113,55 @@ func Delete(name string, flags uintptr) error {
 	return unix.DeleteModule(name, int(flags))
 }
 
+// parallelProbeDep loads a module and it's dependencies
+// Returns once module is loaded.
+func parallelProbeDep(deps depMap, modPath string, params string, opts ProbeOpts) error {
+	dep, present := deps[modPath]
+	if !present {
+		return fmt.Errorf("Module not in depmap %s", modPath)
+	}
+
+	// If it's already loaded, don't try to load again
+	if dep.state == builtin || dep.state == loaded {
```

Comment:
Same here, w.r.t races, IIUC this is maybe all happening in parallel, is this safe

Reply:
This one actually did give me some pause - I do think we can move this safely into the Once